okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/OkHttpClients.java

```diff
@@ -70,18 +70,21 @@
             .build();
     /**
      * The {@link ExecutorService} used for the {@link Dispatcher}s of all OkHttp clients created through this class.
-     * Similar to OkHttp's default, but with two modifications:
+     * Similar to OkHttp's default, but with three modifications:
      * <ol>
      *     <li>A logging uncaught exception handler</li>
      *     <li>
      *         Daemon threads: active request will not block JVM shutdown <b>unless</b> another non-daemon
      *         thread blocks waiting for the result. Most of our usage falls into this category. This allows
      *         JVM shutdown to occur cleanly without waiting a full minute after the last request completes.
      *     </li>
+     *     <li>
+     *         Tracing information is propagated from the caller to the OkHtttp callback.
+     *     </li>
      * </ol>
      */
     private static final ExecutorService executionExecutor =
-            Executors.newCachedThreadPool(executionThreads);
+            Tracers.wrap("OkHttp: callback", Executors.newCachedThreadPool(executionThreads));
```

Comment:
nice

Reply:
Actually, I'm not sure this works correctly. If memory serves, multiple operations may be run from a queue of pending requests here, but we'd end up changing the traceId.

We should add test coverage for queued requests individually retaining correct traceIds.

Reply:
I'm not sure I understand the case you're talking about. Are you say non-callback operations run from the same executor? Or that when requests queue, the traceIds might be off?

Reply:
Two issues:
1. Operations on the dispatcher pool may drain a work queue in their execution, applying thread state more broadly than we expect. Regardless of how it works today, this type of behavior can be added at any time and may cause confusion in our tracing output.
2. When requests are queued, they're not queued on this executor, so the traceIds will not be what we expect.

Reply:
I think this may be similar to some work when trying to instrument a piece of code that uses netty? We also did not want to make netty aware of our Tracing infra. So we created a small executor for each traceId. In here, the RemotingOkhttpCall already captures the traces correctly, so we need to restore them appropriately at this layer? Alternatively, I see AsyncSerializableErrorCallAdapterFactory? If we set and reset the trace in #onResponse and #onFailure, then all the callbacks registered with MoreExecutors.direct should work?